pandas/tests/groupby/transform/test_transform.py

```diff
@@ -1046,6 +1046,16 @@ def test_groupby_transform_with_datetimes(func, values):
     tm.assert_series_equal(result, expected)
 
 
+def test_groupby_transform_dtype():
+    # GH 22243
+    df = DataFrame({"a": [1], "val": [1.35]})
+    result = df.groupby("a")["val"].transform(lambda x: f"+{x}").dtype
+    original = df["val"].dtype
+    expected = df["val"].transform(lambda x: f"+{x}").dtype
+    assert result != original
```

Comment:
Please check the whole DataFrame

Reply:
How does it look this time?

Reply:
Could  you test all cases from the issue?

Reply:
Is that what you mean, I'm not quite sure I fully understand what you mean.

Does this direct listing of `result` and `excepted` need to be further optimized?